ui/src/views/actions/thunks.ts

```diff
@@ -112,8 +115,50 @@ export const getViewForTimeMachine = (
         view,
       })
     )
+  } catch (error) {
+    console.error(error)
+    dispatch(notify(copy.getViewFailed(error.message)))
+    dispatch(setView(cellID, RemoteDataState.Error))
+  }
+}
+
+export const setQueryResultsByQueryID = (queryID: string) => (
+  dispatch,
+  getState: GetState
+): Promise<void> => {
+  try {
+    const state = getState()
+    const files = state.data.queryResultsByQueryID[hashCode(queryID)]
+    if (files) {
+      dispatch(setQueryResults(RemoteDataState.Done, files, null, null))
+      return
+    }
+    dispatch(executeQueries())
+  } catch (error) {
+    // if the files don't exist in the cache, we want to execute the query
+    console.error(error)
+    dispatch(executeQueries())
+  }
+}
+
+export const setQueryResultsForCell = (
+  dashboardID: string,
+  cellID: string,
+  timeMachineID: TimeMachineID
+) => (dispatch, getState: GetState): Promise<void> => {
+  try {
+    dispatch(getViewForTimeMachine(dashboardID, cellID, timeMachineID))
+    const state = getState()
+    const {view} = getActiveTimeMachine(state)
+    const queries = view.properties.queries.filter(({text}) => !!text.trim())
+    const queryID = get(queries, '[0].text', '')
```

Comment:
The function name `setQueryResultsByQueryID` suggests that you will be hashing first and then passing queryID rather than queryText to the function. I think you should be hashing on this line.

Reply:
Definitely. I was thinking that as I was walking through this time around. I'll go ahead and import it in and hash it here